simpl/utils/importing.py

```diff
@@ -0,0 +1,85 @@
+"""Utilities for importing.
+
+The function import_me_maybe() uses the signature
+of the importlib.import_module() function.
+
+Usage:
+
+    from simpl.utils import import_me_maybe
+
+    eventlet = import_me_maybe('eventlet')
+    # this issues a warning (only once) if
+    # eventlet is not found
+
+    # to prevent warnings from ever occurring
+    from simpl.utils import disable_warnings_for
+
+    # this disables the corresponding warnings globally
+    disable_warnings_for('eventlet')
+
+    # this could be in another module
+
+    eventlet = import_me_maybe('eventlet')
+    # produces no warning now
+"""
+import importlib
+import inspect
+import os
+import warnings
+
+from simpl.exceptions import DependencyRequiredWarning
+
+
+def import_me_maybe(name, package=None):
```

Comment:
I've been thinking about adding a `reason` keyword argument here which tells the user why the module is needed. In some cases, we might emit this warning when the requirement is not strictly required. E.g. `eventlet` for the upcoming `server.py`, which I was thinking would run or be importable whether or not eventlet was installed. That might be useful.... `import_me_maybe('eventlet', reason='Required for non-blocking eventlet server')` which would append (or replace?) the generic message built-in and created by the `DependencyRequiredWarning` class.